command/plugin_register.go

```diff
@@ -177,11 +182,13 @@ func (c *PluginRegisterCommand) Run(args []string) int {
 	pluginName := strings.TrimSpace(pluginNameRaw)
 
 	command := c.flagCommand
-	if command == "" && c.flagOCIImage == "" {
-		command = pluginName
+	if c.flagSHA256 != "" {
+		if command == "" && c.flagOCIImage == "" {
```

Comment:
nit: would it be possible to move this conditional to the outer if statement:
`if c.flagSHA256 != "" && (command == "" && c.flagOCIImage == "") {...}`